src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4961,7 +5026,7 @@ std::optional<ProjectionCandidate> MergeTreeData::getQueryProcessingStageWithAgg
     };
 
     auto virtual_block = getSampleBlockWithVirtualColumns();
-    auto add_projection_candidate = [&](const ProjectionDescription & projection)
+    auto add_projection_candidate = [&](const ProjectionDescription & projection, bool normalize_count_not_null = false)
```

Comment:
This flag name looks mysterious to me. What does it really mean?

Reply:
It means not only do we check if `count() == count()`, but also check `count(any not null expression) == count()`